components/tidb_query_datatype/src/codec/table.rs

```diff
@@ -172,29 +179,24 @@ pub fn encode_column_key(table_id: i64, handle: i64, column_id: i64) -> Vec<u8>
     key
 }
 
-/// `decode_handle` decodes the key and gets the handle.
-pub fn decode_handle(encoded: &[u8]) -> Result<i64> {
-    let mut buf = encoded;
-    if buf.read_bytes(TABLE_PREFIX_LEN)? != TABLE_PREFIX {
-        return Err(invalid_type!(
-            "record key expected, but got {}",
-            hex::encode_upper(encoded)
-        ));
-    }
-    buf.read_i64()?;
+/// `decode_int_handle` decodes the key and gets the int handle.
+pub fn decode_int_handle(mut key: &[u8]) -> Result<i64> {
+    check_record_key(key)?;
+    key = &key[PREFIX_LEN..];
+    key.read_i64().map_err(Error::from)
+}
 
-    if buf.read_bytes(RECORD_PREFIX_SEP.len())? != RECORD_PREFIX_SEP {
-        return Err(invalid_type!(
-            "record key expected, but got {}",
-            hex::encode_upper(encoded)
-        ));
-    }
-    buf.read_i64().map_err(Error::from)
+/// `decode_common_handle` decodes key key and gets the common handle.
+pub fn decode_common_handle(mut key: &[u8]) -> Result<&[u8]> {
+    check_record_key(key)?;
+    key = &key[PREFIX_LEN..];
+    Ok(key)
 }
 
+
 /// `truncate_as_row_key` truncate extra part of a tidb key and just keep the row key part.
 pub fn truncate_as_row_key(key: &[u8]) -> Result<&[u8]> {
```

Comment:
when will this function being used (and will it cause problems because it only accepts int handles?)

Reply:
Deleted